src/kernels/execution/cellExecutionMessageHandler.ts

```diff
@@ -418,13 +432,17 @@ export class CellExecutionMessageHandler implements IDisposable {
             ) {
                 if (data.state.msg_id) {
                     // Any future messages sent from `parent_header.msg_id = msg_id` must be handled by the widget with the `mode_id = msg.content.comm_id`.
-                    this.outputsAreSpecificToAWidget = {
+                    this.outputsAreSpecificToAWidget.push({
                         handlingCommId: msg.content.comm_id,
                         msgIdsToSwallow: data.state.msg_id
-                    };
-                } else if (this.outputsAreSpecificToAWidget?.handlingCommId === msg.content.comm_id) {
+                    });
+                } else if (
+                    this.outputsAreSpecificToAWidget.length &&
+                    this.outputsAreSpecificToAWidget[this.outputsAreSpecificToAWidget.length - 1].handlingCommId ===
+                        msg.content.comm_id
```

Comment:
```suggestion
                    this.outputsAreSpecificToAWidget.at(-1)?.handlingCommId ===
                        msg.content.comm_id
```

I was wondering if `at` was a thing yet or not, and seems like it is: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/at looks pretty cool

Reply:
Ohh it depends on TypeScript using es2022